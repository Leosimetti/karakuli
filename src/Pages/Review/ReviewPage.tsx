import React, { useEffect, useState } from 'react'
import { useSelector } from 'react-redux'

import CharacterDisplay_ from '../../Components/CharacterDisplay'
import { NavBar } from '../../Components/NavBar/NavBar'
import Input_ from '../../Components/ReviewInput'
import { selectors } from '../../Data/Slices/userdata'

import { getReviews } from './Review.utils'

import styled from 'styled-components'

const CharacterDisplay = styled(CharacterDisplay_)`
  background-color: #606060;
  margin: 15px auto auto;
`

const Input = styled(Input_)`
  background-color: #606060;
  margin: 15px auto auto;
`
const Wrapper = styled.div`
  height: 100%;
  background-color: #303030;
`
const ArrowRight = styled.span`
  display: flex;

  align-items: center;
  color: #f18c9c;
  user-select: none;
  cursor: pointer;
  font-size: 69px;
  margin: 50px;
`

const ArrowLeft = styled(ArrowRight)`
  transform: rotate(180deg);
`

const Flex = styled.div`
  display: flex;
  justify-content: center;
`

export default function ReviewPage() {
  const userToken = useSelector(selectors.accessToken)

  const review = {
    word: {
      id: -1,
      kanji: '🍆',
      readings: 'BU-BA',
      meaning: 'BEBRA',
      example: 'OH YEAH',
    },
  }

  const [reviews, setReviews] = useState([])
  const [position, setPosition] = useState(0)

  useEffect(() => {
    getReviews(userToken).then((data) => {
      setReviews(data)
    })
  }, [getReviews])

  return (
    <Wrapper>
      <NavBar />
      <Flex>
        <ArrowLeft onClick={() => setPosition(position > 0 ? position - 1 : position)}>➤</ArrowLeft>
        <div>
          <CharacterDisplay review={reviews[position]} />
          <Input callback={alert} status={null} maxLength={20} />
        </div>
        <ArrowRight
          onClick={() => setPosition(position < reviews.length ? position + 1 : position)}
        >
          ➤
        </ArrowRight>
      </Flex>
    </Wrapper>
  )
}
