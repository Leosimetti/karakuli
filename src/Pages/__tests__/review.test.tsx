import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'

import getTestStore from '../../__tests__/testStore'
import ReviewPage from '../Review/ReviewPage'

// eslint-disable-next-line import/no-extraneous-dependencies,node/no-extraneous-import
import { describe, expect, it } from '@jest/globals'
// eslint-disable-next-line node/no-unpublished-import
import { render } from '@testing-library/react'

describe('Auth Page', () => {
  it('renders', () => {
    const { container } = render(
      <BrowserRouter>
        <Provider store={getTestStore()}>
          <ReviewPage />
        </Provider>
      </BrowserRouter>
    )
    expect(container).toMatchSnapshot()
  })
})
