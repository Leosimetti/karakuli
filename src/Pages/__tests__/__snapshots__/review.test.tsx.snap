// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Page renders 1`] = `
.c3 {
  font-family: Roboto,serif;
  font-style: normal;
  font-weight: normal;
  font-size: 55px;
  white-space: nowrap;
  line-height: 55px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #f18c9c;
  text-shadow: 1px 1px 2px red,0 0 1em #f18c9c,0 0 0.2em #f18c9c;
}

.c5 {
  font-size: 40px;
  line-height: normal;
  padding-right: 20px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  background-color: white;
}

.c1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  font-size: 30px;
  line-height: 30px;
}

.c4 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: black;
  margin-top: 10px;
  margin-right: 40px;
}

.c0 {
  height: 100%;
  background-color: #303030;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  font-size: 88px;
  color: green;
}

@media (max-width:800px) {
  .c4 {
    font-size: 24px;
  }
}

@media (max-width:800px) {
  .c2 {
    display: none;
  }
}

<div>
  <div
    class="c0"
  >
    <header
      class="c1"
    >
      <a
        href="/dashboard"
      >
        <div
          class="c2"
        >
          <span
            class="c3"
          >
            カラクリ
          </span>
        </div>
      </a>
      <div
        class="c4"
      >
        <a
          href="/study-lists"
        >
          Каталог
        </a>
        <a
          href="/review"
        >
          Практика
        </a>
        <a
          href="/lessons"
        >
          Уроки
        </a>
        <div
          class="c5"
        >
          Aboba Bebrin
        </div>
      </div>
    </header>
    <div
      class="c6"
    >
      <div
        class="c7"
      />
    </div>
  </div>
</div>
`;
